### PR TITLE
Adding experimental resources

### PR DESCRIPTION
**Pull request by nfx**

### CODE REVIEW

File: catalog/resource_metastore.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewMetastoresAPI(ctx context.Context, m interface{}) MetastoresAPI {
+	return MetastoresAPI{m.(*common.DatabricksClient), ctx}
+}
+
+type MetastoreInfo struct {
+	Name         string  `json:"name"`
```
Comment (alexott) [inline catalog/resource_metastore.go:22]: Should it be force new? Together with storage root?

Comment (nfx) [reply catalog/resource_metastore.go:22]: metastore is identified by UUID, so changing name is just fine. storage root, on the other hand, cannot be updated.

Comment (alexott) [reply catalog/resource_metastore.go:22]: yeah, I saw ForceNew later in the code, after I left the comment. Why not annotation? ;-)

Comment (nfx) [reply catalog/resource_metastore.go:22]: Oh, forgot about it! Will change for annotation.

**Pull request merged**